xilem/src/view/board/kurbo_shape.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implementation of the View trait for various kurbo shapes.
+
+use masonry::widget::KurboShape;
+use vello::kurbo;
+use xilem_core::{AnyElement, AnyView, DynMessage, MessageResult, Mut, OrphanView, SuperElement};
+
+use crate::{Pod, ViewCtx, WidgetView};
+
+pub trait GraphicsView<State, Action = ()>: WidgetView<State, Action, Widget = KurboShape> {}
```

Comment:
I would actually really add a new trait as Element bound here as described in my previous comment instead of enforcing only `KurboShape` like a `SvgElement`, `VectorElement` or `GraphicsElement` trait (though I'm not sure if the name "graphics" is a little bit too general for this kind of use).

So that a `PositionedView` (btw. is the name sufficient, as it also sets the size?) is covered as well with this trait, and potentially in the future something like a `SvgGroup` (`<g>` in SVG) etc.

This would simplify the code in `xilem::view::board`, by not needing `AnyBoardChild` and in masonry as well by avoiding this `Child` enum. This needs an additional (trivial) widget though, but I think that's cleaner and more extensible non-the-less.

Here's a *dirty* sketch of what I mean on the masonry side: https://github.com/Philipp-M/xilem/commit/6571b0aaff59e75c3a0a52eb6ee4b442ee5d0ffe

Feel free to use that code as you like.

Reply:
Co-opted you code. 😄